### ISSUE TITLE
Self-closing tags should always be required in Antlers

### ISSUE THREAD

**Issue by o1y** [statamic/ideas#1275]
Right now, [self-closing tags](https://statamic.dev/antlers#selfclosing-tags) in Antlers aren't strictly enforced. This means that when you use a tag, you don't have to close it, like here `{{ partial:button label="Foo" }}`.

However, if you mix both versions in an Antlers template, it can lead to some incorrect output.

Invalid ❌

```antlers
{{ partial:button label="Foo" }}

{{ partial:button label="Foo" }}
  {{ slot:attributes }}
    foo="Bar"
  {{ /slot:attributes }}
{{/ partial:button }}
```

Correct ✅

```antlers
{{ partial:button label="Foo" /}}

{{ partial:button label="Foo" }}
  {{ slot:attributes }}
    foo="bar"
  {{ /slot:attributes }}
{{/ partial:button }}
```

Since the self-closing version is pretty hidden in the Antlers docs (and isn’t mentioned in the documentation [for the partial tag itself](https://statamic.dev/tags/partial)), figuring out the cause of the faulty rendering in complex templates can take a really long time :)

Either you force self-closing tags through the parser, or make it clearer in the documentation.

**Comment by jasonvarga** [comment]
Forcing tags to be self closing will cause more problems than the few situations where this is an issue.

**Comment by o1y** [comment]
Yeah, maybe then just support both versions or throw an Antlers exception, when you mix between single and pair tags? Right now it’s a bit confusing when you use both versions in your template and then nothing gets rendered.

**Comment by jasonvarga** [comment]
@JohnathonKoster Could that be an option? Throw an exception when there's an ambiguous closing tag situation?

**Comment by JohnathonKoster** [comment]
> @JohnathonKoster Could that be an option? Throw an exception when there's an ambiguous closing tag situation?

This is something we could make work (I would also like this as an option).
